### PR TITLE
[PHP/Dart/Python] Correctly escape strings in single quotes (Fixes #17582)

### LINKED ISSUE #17582: [BUG] PHP/Dart/Python Generator does not escape single quotes in enum values
##### Description

When an enum value contains a single quote, the character is placed directly into the string, causing invalid code in the PHP, Dart, and Python generators.

##### openapi-generator version
7.2.0

##### OpenAPI declaration file content or url
https://gist.github.com/JulianVennen/d291ae93f80f48a687131418aee7da40

##### Generation Details
The command used is
```sh
openapi-generator-cli generate -g php -i openapi.yaml
openapi-generator-cli generate -g dart -i openapi.yaml
openapi-generator-cli generate -g python -i openapi.yaml
```

##### Steps to reproduce
1. Run the PHP generator with the linked spec
2. Open lib/Model/IssueStatus.php in an editor
3. See that the string is not escaped correctly.

##### Suggest a fix
The single quotes should be escaped using backslashes.
This should be done in the escapeText method of the Codegen classes, like it is already done in the AbstractTypeScriptClientCodegen class.

### PR DESCRIPTION
**Pull request by JulianVennen**
This PR fixes #17582 by correctly escaping strings inside the single quotes for the PHP, Dart and Python generators.
For the PHP generator, double quotes escaped by the default generator are unescaped because PHP treats `'\"'` as a string containing a backslash and a double quote.
I have verified that python and dart treat `'\"'` as a string containing just a double quote character.

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] Run the following to [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) and update samples:
  ```
  ./mvnw clean package 
  ./bin/generate-samples.sh ./bin/configs/*.yaml
  ./bin/utils/export_docs_generators.sh
  ``` 
  (For Windows users, please run the script in [Git BASH](https://gitforwindows.org/))
  Commit all changed files. 
  This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. 
  These must match the expectations made by your contribution. 
  You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. 
  IMPORTANT: Do **NOT** purge/delete any folders/files (e.g. tests) when regenerating the samples as manually written tests may be removed.
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master` (upcoming 7.6.0 minor release - breaking changes with fallbacks), `8.0.x` (breaking changes without fallbacks)
- [x] If your PR is targeting a particular programming language, @mention the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) members, so they are more likely to review the pull request.

Dart: @jaumard @josh-burton @amondnet @sbu-WBT @kuhnroyal @agilob @ahmednfwela
PHP: @jebentier, @dkarlovi, @mandrean, @jfastnacht, @ybelenko, @renepardon
Python: @cbornet @tomplus @krjakbrjak @fa0311 @multani

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java [hunk 1]
```diff
@@ -752,10 +752,18 @@ public String toEnumValue(String value, String datatype) {
...
     }
 
+    public String escapeTextInSingleQuotes(String input) {
+        if (input == null) {
+            return null;
+        }
+
+        return escapeText(input).replace("'", "\\'");
+    }
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java:766]: thanks for the PR.

what about putting this method in default codegen instead?

minor suggestion: add a docstring or 1-liner explaining what this function does

Comment (JulianVennen) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractDartCodegen.java]: Sure, I've done both of these

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java [hunk 2]
```diff
@@ -1116,6 +1116,20 @@ public String escapeText(String input) {
...
+     */
+    public String escapeTextInSingleQuotes(String input) {
+        if (input == null) {
+            return null;
+        }
+
+        return escapeText(input).replace("'", "\\'");
+    }
+
+
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:1132]: in abstract typescript codegen, it overrides escapeText to replace `'` with `\\'`

https://github.com/OpenAPITools/openapi-generator/blob/e914c4098b44ac734d3dd21300d433f6cb2974a5/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/AbstractTypeScriptClientCodegen.java#L1068

what about follow the same without introducing `escapeTextInSingleQuotes`?

Comment (JulianVennen) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:1132]: There are several issues with escaping single quotes by default that would also affect other generators:
- Some languages (e.g., PHP) treat `'\"'` as a string containing a backslash and a double quote
- `'` is often used in text like parameter descriptions and would be escaped in comments, which leads to doc blocks like this:
image
- Escaping a single quote is illegal in some formats (e.g., JSON)

Ideally, the context a text is used in (e.g., JSON, single-quoted string, double-quoted string or comment) could be taken into account when a string is escaped, either by making them individual methods or by adding a parameter for that.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/DefaultCodegen.java:1132]: thanks for the explanation. let me think about it and review accordingly

**Comment by wing328** [comment]
thanks again for the PR

let's give it a try

**Pull request merged**